Userland/Applications/PixelPaint/Selection.cpp

```diff
@@ -101,6 +102,15 @@ void Selection::draw_marching_ants(Gfx::Painter& painter, Mask const& mask) cons
     }
 }
 
+void Selection::erase_selection(Gfx::Bitmap& bitmap) const
+{
+    Gfx::Painter painter { bitmap };
+    Gfx::IntRect image_and_selection_intersection = m_editor.image().rect().intersected(bounding_rect());
+    Gfx::IntRect translated_to_layer_space = image_and_selection_intersection.translated(-m_editor.active_layer()->location());
+    painter.clear_rect(translated_to_layer_space, Color::Transparent);
+    m_editor.update();
```

Comment:
I don't know if this would update the `LayerListWidget`, since that relies on the `ImageClient::image_did_modify_layer_bitmap`. I'm thinking perhaps a better API would be `Layer::erase_selection(Selection const&)`, since that can properly broadcast changes everywhere by calling `did_modify_bitmap` (which can only be called from `Layer`)